paymentsheet/src/main/java/com/stripe/android/paymentsheet/state/PaymentElementLoader.kt

```diff
@@ -408,56 +409,38 @@ internal class DefaultPaymentElementLoader @Inject constructor(
         customer: CustomerInfo?,
         initializationMode: PaymentElementLoader.InitializationMode
     ): LinkState? {
-        return if (configuration.allowsLink && elementsSession.isLinkEnabled) {
-            loadLinkState(
+        val linkConfig =
+            createLinkConfiguration(
                 configuration = configuration,
                 customer = customer,
                 elementsSession = elementsSession,
-                merchantCountry = elementsSession.merchantCountry,
-                passthroughModeEnabled = elementsSession.linkPassthroughModeEnabled,
-                linkSignUpDisabled = elementsSession.disableLinkSignup,
-                useAttestationEndpointsForLink = elementsSession.useAttestationEndpointsForLink,
-                suppress2faModal = elementsSession.suppressLink2faModal,
-                flags = elementsSession.linkFlags,
                 initializationMode = initializationMode
-            )
-        } else {
-            null
-        }
+            ) ?: return null
```

Comment:
Nit: We could do a `createLinkConfiguration(…)?.let { configuration -> … }` to avoid the elvis-operator return.